prusti-tests/tests/verify/fail/arrays-slices/out-of-bounds.rs

```diff
@@ -0,0 +1,8 @@
+use prusti_contracts::*;
+
+fn main() {}
+
+
+fn third(s: &[i32]) -> i32 {
+    s[2]  //~ ERROR index out of bounds
```

Comment:
I would add more text from the error message to check that Prusti says that the statement *may* fail with an index out of bounds.

Reply:
updated the error message.. it's a bit ugly, because i have to include the double quotes now, as the full message is `assertion might fail with ""index out of bounds: the length is {} but the index is {}", move _3, _2"`, but for testing it's clearer now i think.

Reply:
Uh, didn't you fix the "ugliness"? Is the fix outside of this PR?

Reply:
Also it's a little strange that the message says "assertion" when there is no assertion as far as the user is concerned.

Reply:
@fpoli what we fixed was a panic/`unreachable!()` when displaying an `AssertKind::BoundsCheck`, the message is now what `fmt_assert_args` prints.
@Aurel300 i think i have to take back what i said in today's meeting; the "assertion might fail with" part comes from [prusti-viper/src/encoder/errors/error_manager.rs:273](https://github.com/viperproject/prusti-dev/blob/master/prusti-viper/src/encoder/errors/error_manager.rs#L273), so we can only really influence what is inside the quotes without extra effort. viper only reports an assertion failure as well, it obviously cannot tell that this is inside a bounds check.

Reply:
Ok, I see. However, in the file you link, the `ErrorCtxt::AssertTerminator(...)` is actually something Prusti adds when registering the assertion failure with the error manager. Can we easily (i.e. hopefully without string matching…) detect that a particular assert in MIR corresponds to a bound check and if so, register it with a different `ErrorCtxt`?
This might be a lot of work for a tiny annoyance, but it's worth checking.

Reply:
I think changing "assertion might fail with" to "statement might fail with" would be an improvement for all `AssertTerminator`'s usages.

Reply:
@Aurel300 hm, good idea, added that :-)
@fpoli i guess that general change is something for a different PR